omnipaxos/src/storage.rs

```diff
@@ -704,14 +712,25 @@ where
     pub(crate) fn create_diff_snapshot(
         &mut self,
         from_idx: u64,
-    ) -> StorageResult<SnapshotType<T>> {
-        if self.get_compacted_idx() >= from_idx {
-            Ok(SnapshotType::Complete(self.create_decided_snapshot()?))
+    ) -> StorageResult<Option<SnapshotType<T>>> {
+        let log_decided_idx = match self.stopsign_is_decided() {
+            true => self.get_decided_idx() - 1,
+            false => self.get_decided_idx(),
+        };
+        let compacted_idx = self.get_compacted_idx();
+        if from_idx <= compacted_idx {
+            if compacted_idx < log_decided_idx {
+                Ok(Some(SnapshotType::Complete(
+                    self.create_snapshot(log_decided_idx)?,
+                )))
+            } else {
```

Comment:
This is an optimization because when `compacted_idx == log_decided_idx` our current snapshot is up to date and we can just use it. 

However, it also necessary for when an "empty" snapshot doesn't  act like an identity element, like with `LatestValue` in the tests. This is because if we call `create_snapshot` when our log consists of entirely a snapshot, we will it will merge an empty snapshot onto the current one.